### DISCUSSION TITLE
2FA for Remote Access

### DISCUSSION THREAD

**Discussion by Wizlonex** [lxdware/lxd-dashboard#54] [Ideas]
I am not a coder, so I have no idea how difficult this feature request is.

I'd like to give remote access to users.  Presently I do this via a VPN connection to my lan, then direct access via http:/ over the lan.  It works.  I think it's pretty secure, but it's not convenient for folks to use a vpn all the time.

I can easily make the site https-only access of course e.g. using certbot.  However, given that lxdware has essentially root access to the lxd servers, whereinstances can be copied/deleted as well as console-accessed, it would be highly desirable to be able to enable 2FA access e.g. via google authenticator so that we're not just relying on a single point of failure - a password (which can even be a WEAK password at that) - even for lan access.

Just wondering how hard that would be, and whether it creates more problems than it solves?

Love your lxdware!!  THANK YOU.

Very Respectfully,

Andrew

**Comment by matthewalanpenning** [comment]
Thank you Andrew! This is a great idea and something I am planning to implement in my next major release. I am currently in the process of of converting the backend to python and have a solution in mind to allow for 2FA. It is still several months away, but I have already started looking into adding additional authentication options.

**Comment by erik78se** [comment]
I can show you how to use wireguard to get a peer vpn up to do this.

Den tors 22 dec. 2022 20:22Matthew Penning ***@***.***> skrev:
